src/v/transform/logging/rpc_client.cc

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "rpc_client.h"
+
+#include "cluster/topic_table.h"
+#include "model/namespace.h"
+#include "transform/logging/logger.h"
+#include "transform/logging/record_batcher.h"
+
+#include <seastar/core/future.hh>
+
+namespace transform::logging {
+
+rpc_client::rpc_client(
+  rpc::client* rpc_client, cluster::topic_table* topic_table)
+  : _rpc_client(rpc_client)
+  , _topic_table(topic_table) {}
+
+ss::future<errc>
+rpc_client::write(model::partition_id pid, io::json_batches batches) {
+    model::topic_namespace_view ns_tp{model::transform_log_internal_nt};
+    // TODO(oren): is topic metadata cache OR topic table more preferable here?
+    auto cfg = _rpc_client->find_topic_cfg(ns_tp);
+    if (!cfg.has_value()) {
+        co_return errc::topic_not_found;
+    }
+
+    // topic should reflect cluster config at time of creation, so we can assume
+    // batch_max_bytes it's set
+    auto max_batch_size = cfg->properties.batch_max_bytes;
+    vassert(
+      max_batch_size.has_value(),
```

Comment:
>     // topic should reflect cluster config at time of creation, so we can assume
    // batch_max_bytes it's set

not sure i follow why this implies that the property has a value

Reply:
>not sure i follow...

that's because it's nonsense...not sure how I arrived there either 😅